pkg/storkctl/snapshot.go

```diff
@@ -220,3 +223,153 @@ func getSnapshotStatusAndTime(snap *snapv1.VolumeSnapshot) (string, string) {
+	var snapGroup bool
+
+	restoreSnapshotCommand := &cobra.Command{
+		Use:     restoreSubCommand,
+		Aliases: []string{"restore"},
+		Short:   "Restore snapshot to source PVC",
+		Run: func(c *cobra.Command, args []string) {
+			if len(args) != 1 {
+				util.CheckErr(fmt.Errorf("exactly one argument needs to be provided for volumesnapshotrestore name"))
+				return
+			}
+			restoreCRDName := args[0]
+			snapRestore := &storkv1.VolumeSnapshotRestore{
+				Spec: storkv1.VolumeSnapshotRestoreSpec{
+					SourceName:      snapName,
+					SourceNamespace: snapNamespace,
+					GroupSnapshot:   snapGroup,
+				},
+			}
+			snapRestore.Name = restoreCRDName
+			snapRestore.Namespace = cmdFactory.GetNamespace()
+			_, err := k8s.Instance().CreateVolumeSnapshotRestore(snapRestore)
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+
+			msg := fmt.Sprintf("Snapshot %v restored successfully", restoreCRDName)
+			printMsg(msg, ioStreams.Out)
+		},
+	}
+	restoreSnapshotCommand.Flags().StringVarP(&snapName, "snapname", "n", "", "Snapshot name to be restored")
+	restoreSnapshotCommand.Flags().StringVarP(&snapNamespace, "namespace", "", "default", "Namespace of snapshot")
+	restoreSnapshotCommand.Flags().BoolVarP(&snapGroup, "is-groupsnapshot", "g", false, "True if snapshot is group, default false")
+	return restoreSnapshotCommand
+}
+
+func newGetVolumeSnapshotRestoreCommand(cmdFactory Factory, ioStreams genericclioptions.IOStreams) *cobra.Command {
+	restoreSnapshotCommand := &cobra.Command{
+		Use:     restoreSubCommand,
+		Aliases: []string{"restore"},
+		Short:   "Get snapshot restore status",
+		Run: func(c *cobra.Command, args []string) {
+			var snapRestoreList *storkv1.VolumeSnapshotRestoreList
+			var err error
+
+			namespaces, err := cmdFactory.GetAllNamespaces()
+			if err != nil {
+				util.CheckErr(err)
+				return
+			}
+
+			if len(args) > 0 {
+				snapRestoreList = new(storkv1.VolumeSnapshotRestoreList)
+				for _, restoreName := range args {
+					for _, ns := range namespaces {
+						snapRestore, err := k8s.Instance().GetVolumeSnapshotRestore(restoreName, ns)
+						if err != nil {
+							util.CheckErr(err)
+							return
+						}
+						snapRestoreList.Items = append(snapRestoreList.Items, *snapRestore)
+					}
+				}
+			} else {
+				var tempRestoreList storkv1.VolumeSnapshotRestoreList
+				for _, ns := range namespaces {
+					snapRestoreList, err = k8s.Instance().ListVolumeSnapshotRestore(ns)
+					if err != nil {
+						util.CheckErr(err)
+						return
+					}
+					tempRestoreList.Items = append(tempRestoreList.Items, snapRestoreList.Items...)
+				}
+				snapRestoreList = &tempRestoreList
+
+			}
+
+			if len(snapRestoreList.Items) == 0 {
+				handleEmptyList(ioStreams.Out)
+				return
+			}
+
+			if err := printObjects(c, snapRestoreList, cmdFactory, snapRestoreColumns, snapshotRestorePrinter, ioStreams.Out); err != nil {
+				util.CheckErr(err)
+				return
+			}
+		},
+	}
+
+	return restoreSnapshotCommand
```

Comment:
Need to call `BindGetFlags()`, check `migration.go`. Also need a UT with `--all-namespaces`